### ISSUE TITLE
Support Multiple Tag Definitions in ComponentDefinition

### ISSUE THREAD

**Issue by JRJurman** [Tram-One/tram-lite#23]
## Summary

A common pattern we've seen is that developers will define multiple components all at once, wanting to use a single `<template is="component-definition">`.

Example:
```html
<template is="component-definition">
  <nginx-log-line>
    <!-- first component definition -->
  </nginx-log-line>
  <appserver-log-line>
    <!-- second component definition -->
  </appserver-log-line>
  <appserver-msg-line>
    <!-- third component definition -->
  </appserver-msg-line>
</template>
```

Right now, we only process the first element child, but there's not a technical reason we couldn't iterate through each element child, and define a new component for each.

**Comment by JRJurman** [comment]
If nothing else, if we can't support this, we should throw a warning or error if this isn't possible, and more than one child element is found!

**Comment by JRJurman** [comment]
This was released as part of v3.1.0
